### PR TITLE
Rewrite KeepAlive logic to be more in line with the Notchian server.

### LINKED ISSUE #291: Rewrite keep alive logic
As it currently stands, Obsidian can handle the login and play client states, keeping the player online through the keep alive. While the keep-alive logic isn't "broke" per se, it's incorrectly implemented:

> The server will frequently send out a keep-alive, each containing a random ID. The client must respond with the same payload (see serverbound Keep Alive). If the client does not respond to them for over 30 seconds, the server kicks the client. Vice versa, if the server does not send any keep-alives for 20 seconds, the client will disconnect and yields a "Timed out" exception.

> The Notchian server uses a system-dependent time in milliseconds to generate the keep alive ID value. 

https://github.com/ObsidianMC/Obsidian/blob/c38662b4f35e979a4584a7184ab82cd1acc7a211/Obsidian/Client.cs#L468-L470

https://github.com/ObsidianMC/Obsidian/blob/c38662b4f35e979a4584a7184ab82cd1acc7a211/Obsidian/Server.cs#L663

If we take a look here at these two code examples, we can see that the method takes a `long` as [the protocol states](https://wiki.vg/Protocol#Keep_Alive_.28clientbound.29), however, it will cast the result of the subtraction to an `int`. Under perfect circumstances, this will not be an issue as the server ensures that the ids generated will always remain below `1000` and the client is intended to return the id received. However, through the use of mods, the id can be modified and return a number between `int.MaxValue` and `long.MaxValue`, which will cause an exception to be thrown. It is speculated that the server will catch the exception and handle it silently, however in the event that it does not, we can consider this to be an attack vector.

I believe the correct solution is to use Unix timestamps with millisecond precision (`DateTimeOffset.Now.ToUnixTimeMilliseconds()`) for the keep-alive id. A unix timestamp still fits in the range of `long`, prevents duplicate packet ids, and still allows us to track the player's ping.

### PR DESCRIPTION
**Pull request by Naamloos**
This PR would close #291.

### CODE REVIEW

**Review by Tides — CHANGES_REQUESTED** [review]
We should make a config value to set the keep alive timeout and pass it down to the client so people can configure it. Beside that LGTM ~

**Comment by Naamloos** [comment]
To whoever decides to do the second review: please wait before merging, I want to add the config value(s).

File: Obsidian/Client.cs
Comment (Craftplacer) [inline Obsidian/Client.cs]: small code style suggestion: use 30_000 instead of 30000, in order to make it more readable

File: Obsidian/Server.cs
Comment (Craftplacer) [inline Obsidian/Server.cs]: Make this magic number is a constant field perhaps? (comment could be moved alongside it)

**Review by Craftplacer — DISMISSED** [review]

**Comment by Naamloos** [comment]
Requested changes were made. Asking for re-review

**Review by Tides — APPROVED** [review]

**Review by Craftplacer — APPROVED** [review]

**Pull request merged**
